docs/guides/integrating-cert-manager.md

```diff
@@ -0,0 +1,290 @@
+# Automating Secure Server-Client Communication using Cert-Manager
+
+Securing communication between an application and its clients is a crucial part of modern application architectures. One
+of the most important steps in this process is implementing HTTPS (HTTP over TLS/SSL) for all communication between your
+server and clients. This encrypts the data transmitted between the two, preventing eavesdropping and tampering. To do
+this, you need an SSL/TLS certificate from a trusted Certificate Authority (CA). However, issuing and managing
+certificates can be a complicated manual process. Luckily, there are many services and tools available to simplify and
+automate certificate issuance and management.
+
+This guide will demonstrate how to configure HTTPS for your application using Gateway resources managed by NGINX
+Kubernetes Gateway (NKG); [Let’s Encrypt](https://letsencrypt.org) as the Certificate Authority (CA) issuing the
+certificates used in securing these resources; and [cert-manager](https://cert-manager.io) to automate the provisioning
+and management of these certificates.
+
+## Prerequisities
+
+1.	Administrator access to a Kubernetes cluster is required.
+2.	[Helm](https://helm.sh) and [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) must be installed locally.
+3.	Deploy NGINX Kubernetes Gateway (NKG) following the [deployment instructions](../installation.md).
+4.	A DNS resolvable domain name is required. It must resolve to the public endpoint of the NKG deployment, and this
+    public endpoint must be an external IP address or alias accessible over the internet. The process here will depend
+    on your DNS provider. This DNS name will need to be resolvable from the Let’s Encrypt servers, which may require
+    that you wait for the record to propagate before it will work.
+
+## Overview
+
+![cert-manager ACME Challenge and certificate managment with Gateway API](/docs/images/cert-manager-gateway-workflow.png)
```

Comment:
for the diagram:
- Could we use different colors or patterns for the lines trough which the traffic goes and for the ones that represent relationship between resources?
- Could we also add a temp HTTPRoute resource that cert-manager creates? End also make the traffic from Let's Encrypt go through NKG and to cert-manager (I believe this how it works)?

Reply:
I'm really bad at diagrams, but sure, I'll improve this 😅

Reply:
I've updated the diagram, feedback welcome 😁!